src/util.ts

```diff
@@ -249,3 +249,13 @@ function lastDotCmp(s: string): string {
 	const cmps = s.split('.');
 	return cmps[cmps.length - 1];
 }
+
+/**
+ * Strips file part (if any) from container name (last component of container path)
+ * For example TS may return the following name: /node_modules/vscode-jsonrpc/lib/cancellation.
+ * We consider that if name contains path separtor then container name is empty
+ * @param containerName
+ */
+function stripFileInfo(containerName: string): string {
```

Comment:
I think you are right, we cannot guarantee that a file defining the symbol corresponds to the file referencing the symbol in any way: For TS projects, it can be in a different folder (src vs lib) and it is a different file (.ts vs .d.ts vs .js). For JS projects, the typings are probably in a single index.d.ts on DefinitelyTyped.

Leaving the path out will make symbol matching very fuzzy, but I don't see a different way.